cardinal/world.go

```diff
@@ -263,32 +276,36 @@ func RegisterSystems(w *World, systems ...System) {
 }
 
 func RegisterComponent[T metadata.Component](world *World) error {
-	return ecs.RegisterComponent[T](world.implWorld)
+	return ecs.RegisterComponent[T](world.instance)
 }
 
 // RegisterMessages adds the given messages to the game world. HTTP endpoints to queue up/execute these
 // messages will automatically be created when StartGame is called. This Register method must only be called once.
 func RegisterMessages(w *World, msgs ...AnyMessage) error {
-	return w.implWorld.RegisterMessages(toMessageType(msgs)...)
+	return w.instance.RegisterMessages(toMessageType(msgs)...)
 }
 
 // RegisterQueries adds the given query capabilities to the game world. HTTP endpoints to use these queries
 // will automatically be created when StartGame is called. This Register method must only be called once.
 func RegisterQueries(w *World, queries ...AnyQueryType) error {
-	return w.implWorld.RegisterQueries(toIQueryType(queries)...)
+	return w.instance.RegisterQueries(toIQueryType(queries)...)
+}
+
+func (w *World) Instance() *ecs.World {
```

Comment:
This is meant to only be used in tests, correct? I'd suggest renaming this to TestInstance or TestToECSWorld, and adding an appropriate comment.

Reply:
Not necessarily only for test, but I'll just call it InternalInstance to make it clear that people should typically not use this

Reply:
you know what, let's make a separate ticket for this. the cascading rebase i have to do for this is not very pleasant.